docs/executor_tutorial/azure_batch.rst

```diff
@@ -0,0 +1,255 @@
+in this environment. Should they be missing install with:
+
+.. code:: console
+
+   pip install azure-batch
+   pip install azure-storage-blob
+
+In addition you will need the
+`Azure CLI command <https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest>`__ 
+installed.
+
+Create an Azure storage account and upload example data
+:::::::::::::::::::::::::::::::::::::::::::::::
+
+We will be starting from scratch, i.e. we will 
+create a new resource group and storage account. You can obviously reuse 
+existing resources instead.
+
+.. code:: console
+
+   # change the following names as required
+   # azure region where to run:
+   region=westus
+
+   # name of the resource group to create:
+   resgroup=snakemaks-rg
+
+   # name of storage account to create (all lowercase, no hyphens etc.):
+   stgacct=snakemaksstg
+
+   # create a resource group with name and in region as defined above
+   az group create --name $resgroup --location $region
+
+   # create a general purpose storage account with cheapest SKU
+   az storage account create -n $stgacct -g $resgroup --sku Standard_LRS -l $region
+
+Get a key for that account and save it as ``stgkey`` for later use:
+
+.. code:: console
+
+   stgkey=$(az storage account keys list -g $resgroup -n $stgacct | head -n1 | cut -f 3)
+
+Next, you will create a storage container (think: bucket) to upload the Snakemake tutorial data to:
+
+.. code:: console
+
+   az storage container create --resource-group $resgroup --account-name $stgacct \
+       --account-key $stgkey --name snakemake-tutorial
+
+   cd /tmp
+
+   git clone https://github.com/snakemake/snakemake-tutorial-data.git
+
+   cd snakemake-tutorial-data
+
+   az storage blob upload-batch -d snakemake-tutorial --account-name $stgacct \
+       --account-key $stgkey -s data/ --destination-path data
+
+We are using `az storage blob` for uploading, because that `az` is already installed.
+Another cli tool for uploading to azure storage is 
+`azcopy <https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10>`__.
+
+Create an Azure Batch Account
+:::::::::::::::::::::::::::::::::::::::::
+
+Create a new azure batch account. The batch account key will be given to snakemake to enable creation of batch resources by snakemake.
+
+.. code:: console
+
+    # can set variables as appropriate
+    resgroup = snakemake-rg
+    accountname = snakebatch01
+    location = westus
+    az batch account create --resource-group $resgroup --name $accountname --location $location\
+
+
+.. code:: console
+
+    az_batch_account_key=$(az batch account keys list --resource-group $resgroup --name $accountname -o tsv | head -n1 | cut -f2)
+
+
+
+To run the test workflow, two primary environment variables need to be set local to the snakemake invocation. The azure batch account key, and the azure storage account url with an SAS key.
+
+.. code:: console
+
+     export AZ_BLOB_ACCOUNT_URL='${storage_account_url_with_sas}'
```

Comment:
Do you want users to generate this in CLI or by Portal?

Reply:
CLI would be nice but I can't seem to find the command to generate an SAS url for a storage account that produces the full URI.

Reply:
export sastoken=$(az storage account generate-sas \
    --account-name $stgacct \
    --permissions acdlrw \
    --services bf \
    --resource-types co \
    --expiry $expiry_date)

export sas_url=$(az storage account show-connection-string \
    -g $resgroup \
    -n $stgacct \
    --protocol https \
    -o tsv | cut -f5,9 -d ';' | cut -f 2 -d '=')

 export storage_account_url_with_sas='${sas_url}${sastoken}'

Something like this. can be made better.